### PR TITLE
Chores: bump up f8 dep

### PR DESCRIPTION
**Pull request by stanlyDoge**
Signed-off-by: Stanislav Knot <sknot@redhat.com>

### Type of change
- Bugfix
- Enhancement

### Description
Upgrade fabric8 dependency.

Memory resource changes in tests: It seems getting `getSizeLimit().getAmount()` was fixed in fabric8 because it was returning (e.g.) `4Gi`. It is obvious the amount is `4` (not `4Gi` as before, `Gi` is the format (`getSizeLimit().getFormat()`).

The bug with detecting java and enabling http2 has been fixed.

I had to do some changes (random resource name) in ITs because of caching resources which caused some flakiness.

### Checklist
- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -72,9 +72,9 @@
         <maven-jar-plugin.version>3.1.0</maven-jar-plugin.version>
         <sundrio.version>0.19.0</sundrio.version>
 
-        <fabric8.kubernetes-client.version>4.6.4</fabric8.kubernetes-client.version>
-        <fabric8.openshift-client.version>4.6.4</fabric8.openshift-client.version>
-        <fabric8.kubernetes-model.version>4.6.4</fabric8.kubernetes-model.version>
+        <fabric8.kubernetes-client.version>4.10.0</fabric8.kubernetes-client.version>
```
Comment (dalelane) [inline pom.xml:75]: worth getting 4.10.1 ?

looks like there was a regression in the 4.10.0 release https://github.com/fabric8io/kubernetes-client/issues/2189 that was addressed in a new release today https://github.com/fabric8io/kubernetes-client/releases/tag/v4.10.1

Comment (stanlyDoge) [reply pom.xml:75]: Yeah, in my local branch I have bumped to 4.10.1 but there is some build issue (you can take a look at travis build logs).

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

File: .azure/templates/setup_java.yaml [hunk 2]
```diff
@@ -21,6 +21,16 @@ steps:
...
+- bash: |
+    echo "##vso[task.setvariable variable=JAVA_VERSION]11"
+  displayName: 'Setup JAVA_VERSION=11'
+  condition: eq(variables['JDK_VERSION'], '11')
+
+- bash: |
+    echo "##vso[task.setvariable variable=JAVA_VERSION]1.8"
+  displayName: 'Setup JAVA_VERSION=1.8'
+  condition: eq(variables['JDK_VERSION'], '1.8')
+
```
Comment (scholzj) [inline .azure/templates/setup_java.yaml:33]: Can you explain what this does and why is it needed?

File: .travis.yml [hunk 3]
```diff
@@ -10,6 +10,7 @@ jdk:
 services:
 - docker
 before_install:
+- if [ $(java -version 2>&1 | sed -E -n 's/.* version "([0-9]*).*$/\1/p') -gt 1 ] ; then export JAVA_VERSION=$(java -version 2>&1 | sed -E -n 's/.* version "([0-9]*).*$/\1/p'); else export JAVA_VERSION=$(java -version 2>&1 | sed -E -n 's/.* version "([0-9]*)\.([0-9]).*$/\1.\2/p'); fi;
```
Comment (scholzj) [inline .travis.yml:13]: Same as above - Can you explain what this does and why is it needed?

File: Jenkinsfile-pr [hunk 4]
```diff
@@ -37,6 +37,7 @@ pipeline {
         TEST_CLUSTER_ADMIN = "admin"
         OPERATOR_IMAGE_PULL_POLICY = "IfNotPresent"
         COMPONENTS_IMAGE_PULL_POLICY = "IfNotPresent"
+        JAVA_VERSION = "11"
```
Comment (scholzj) [inline Jenkinsfile-pr:40]: Why is Jenkins expected to do Java 11?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java [hunk 5]
```diff
@@ -2789,6 +2790,7 @@ StatefulSet getKafkaStatefulSet()   {
         }
 
         @SuppressWarnings("deprecation")
+        @SuppressFBWarnings(value = "UPM_UNCALLED_PRIVATE_METHOD")
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:2793]: Why is this needed and wasn't needed before?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java [hunk 6]
```diff
@@ -323,7 +324,7 @@ public static int getPodGeneration(Pod resource) {
             long pollingIntervalMs = 1_000;
             long timeoutMs = operationTimeoutMs;
 
-            operation().inNamespace(namespace).withName(name).cascading(cascading).withGracePeriod(-1L).delete();
+            operation().inNamespace(namespace).withName(name).withPropagationPolicy(cascading ? DeletionPropagation.FOREGROUND : DeletionPropagation.BACKGROUND).delete();
```
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java:327]: Are you sure this is right? `DeletionPropagation.BACKGROUND` does not sound like it is the same as `cascading(false)`. I think that `DeletionPropagation.ORPHAN` sounds more like it.

Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/StatefulSetOperator.java:327]: Have you double checked that the gracePeriod is still maintained from the pod when the `withGracePeriod` is removed?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java [hunk 7]
```diff
@@ -349,7 +349,7 @@ public void testGenerateStatefulSetWithEmptyStorageSelector() {
 
     @Test
     public void testGenerateStatefulSetWithSetSizeLimit() {
-        String sizeLimit = "1Gi";
+        String sizeLimit = "1";
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java:352]: This doesn't sound right to me. Why is this needed nd why is this ok to change it?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectS2IClusterTest.java [hunk 8]
```diff
@@ -284,7 +284,7 @@ public void testGenerateBuildConfig() {
         assertThat(bc.getSpec().getSuccessfulBuildsHistoryLimit(), is(new Integer(5)));
         assertThat(bc.getSpec().getFailedBuildsHistoryLimit(), is(new Integer(5)));
         assertThat(bc.getSpec().getResources().getLimits().get("cpu").getAmount(), is("42"));
-        assertThat(bc.getSpec().getResources().getRequests().get("mem").getAmount(), is("4Gi"));
+        assertThat(bc.getSpec().getResources().getRequests().get("mem").getAmount(), is("4"));
```
Comment (scholzj) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaConnectS2IClusterTest.java:287]: Again, this does not seem to be right. How does this impact the behaviour of the public APIs where you use 4Gi today?

File: kafka-agent/pom.xml [hunk 9]
```diff
@@ -29,6 +29,11 @@
             <version>1.7.25</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>com.github.spotbugs</groupId>
```
Comment (scholzj) [inline kafka-agent/pom.xml:33]: AFAIK this should be optional or provided to not mix into the runtime binaries.

File: operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java [hunk 10]
```diff
@@ -103,7 +103,7 @@ private static OkHttpClient getOkHttpClient(KubernetesClient client)   {
 
         if (kubernetesVersion != null) {
             try {
-                futureVersion = Future.succeededFuture(new VersionInfo(Util.parseMap(kubernetesVersion)));
+                futureVersion = Future.succeededFuture(Util.parseVersionInfo(kubernetesVersion));
```
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java:106]: What is the value of having this in Util and not right in this class? It does not seem to be used from other places.

File: operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperator.java [hunk 11]
```diff
@@ -173,7 +173,7 @@ public AbstractNonNamespacedResourceOperator(Vertx vertx, C client, String resou
 
     protected Future<ReconcileResult<T>> internalPatch(String name, T current, T desired, boolean cascading) {
         try {
-            T result = operation().withName(name).cascading(cascading).patch(desired);
+            T result = operation().withName(name).patch(desired);
```
Comment (scholzj) [inline operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperator.java:176]: Did you checked why we added the cascading here?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java [hunk 12]
```diff
@@ -105,6 +106,7 @@ public void testCreateModifyDelete(VertxTestContext context)    {
             }))
             .compose(rr -> op.reconcile(namespace, RESOURCE_NAME, null))
             .setHandler(context.succeeding(rrDeleted -> {
+                TestUtils.waitFor("resouce deleted", 1_000, 30_000, () -> op.get(namespace, RESOURCE_NAME) == null);
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java:109]: Why do we have to wait here now and did not had to wait before? That sounds like it will break whatever place is using it and is expecting the old behavior.

File: operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java (hunk 10 above)
Comment (stanlyDoge) [reply operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java:106]: It was there before o.O

Comment (scholzj) [reply operator-common/src/main/java/io/strimzi/operator/PlatformFeaturesAvailability.java:106]: Before you used a method from Util. But it was generic method used on many places. Now you added your own method (which is fine) used only in here and in the related test. So it does not mke sense anymore I guess?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java (hunk 12 above)
Comment (stanlyDoge) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java:109]: It is flaky now but I am afraid it is the same as https://github.com/strimzi/strimzi-kafka-operator/pull/1424/files#diff-210c63328d6cf1d8f3f414fdd5244bf8R129

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java [hunk 13]
```diff
@@ -277,18 +277,17 @@ public void setup(VertxTestContext testContext) throws InterruptedException {
         kafkaConnectOperator.createWatch(NAMESPACE, e -> testContext.failNow(e))
             .onComplete(testContext.succeeding())
             .compose(watch -> {
-                kafkaConnectS2iOperator = new KafkaConnectS2IAssemblyOperator(vertx, pfa, ros, config, x -> api);
+                kafkaConnectS2iOperator = new KafkaConnectS2IAssemblyOperator(vertx,
+                    pfa,
+                    ros,
+                    config,
+                    x -> api);
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java]: Can we rename `x` to something a little more descriptive?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java (hunk 5 above)
Comment (tombentley) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java:2793]: How was this resolved?

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/ResourceUtils.java [hunk 14]
```diff
@@ -456,6 +456,7 @@ public static KafkaConnectS2I createEmptyKafkaConnectS2ICluster(String clusterCm
                 .withNamespace(clusterCmNamespace)
                 .withLabels(TestUtils.map(Labels.KUBERNETES_DOMAIN + "part-of", "tests",
                         "my-user-label", "cromulent"))
+                .withAnnotations(emptyMap())
```
Comment (tombentley) [inline cluster-operator/src/test/java/io/strimzi/operator/cluster/ResourceUtils.java:459]: Why is this now necessary?

File: mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java [hunk 15]
```diff
@@ -126,7 +128,7 @@ private void mockScale(String resourceName, RollableScalableResource<StatefulSet
 
     @Override
     protected void mockCreate(String resourceName, RollableScalableResource<StatefulSet, DoneableStatefulSet> resource) {
-        when(resource.create(any())).thenAnswer(cinvocation -> {
+        when(resource.create(any(StatefulSet.class))).thenAnswer(cinvocation -> {
```
Comment (tombentley) [inline mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java:131]: So I'm guessing this is necessary because there's some new overloaded `create()` method which makes the old `any()` ambiguous. Can you tell me what the new `create()` method is?

File: topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorBaseIT.java [hunk 16]
```diff
@@ -566,17 +567,17 @@ protected void waitForEvent(KafkaTopic kafkaTopic, String expectedMessage, Topic
...
             return filtered.stream().anyMatch(event ->
-                    Pattern.matches(expectedMessage, event.getMessage()) &&
+                    Pattern.matches(expectedMessage, event.getReason()) &&
                         Objects.equals(expectedType.name, event.getType()) &&
-                        event.getInvolvedObject() != null &&
-                        event.getLastTimestamp() != null &&
-                        Objects.equals("KafkaTopic", event.getInvolvedObject().getKind()) &&
-                        Objects.equals(kafkaTopic.getMetadata().getName(), event.getInvolvedObject().getName()));
+                        event.getRegarding() != null &&
+                        event.getDeprecatedLastTimestamp() != null &&
```
Comment (tombentley) [inline topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorBaseIT.java:578]: If it's deprecated should we still be using it?

File: mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java (hunk 15 above)
Comment (stanlyDoge) [reply mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java:131]: I believe it is caused by adding this: https://github.com/fabric8io/kubernetes-client/blob/master/kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/Createable.java#L22

File: topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorBaseIT.java (hunk 16 above)
Comment (stanlyDoge) [reply topic-operator/src/test/java/io/strimzi/operator/topic/TopicOperatorBaseIT.java:578]: I am not sure how important it is in TO. If you think it is not needed anymore, I can remove it.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java (hunk 13 above)
Comment (stanlyDoge) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java]: Not actually my code but sure.

File: cluster-operator/src/test/java/io/strimzi/operator/cluster/ResourceUtils.java (hunk 14 above)
Comment (stanlyDoge) [reply cluster-operator/src/test/java/io/strimzi/operator/cluster/ResourceUtils.java:459]: It seems to be `null` by default. Some tests were relying on non null value. See https://github.com/strimzi/strimzi-kafka-operator/blob/f983bb0fbbab7ce444c123d0eb8d14aaea1e14f2/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectS2IAssemblyOperatorTest.java#L1023

File: mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java (hunk 15 above)
Comment (tombentley) [reply mockkube/src/main/java/io/strimzi/test/mockkube/StatefulSetMockBuilder.java:131]: Ah, excellent, that's a method I requested myself actually. In the `AbstractResourceOperator` (and similar) we used to get "generic array creation" warnings because there was only a varargs method in `Creatable`. This means that we won't get those warnings now, so the `@SuppressWarnings` won't be needed any more. Can you hunt down those annotations and remove them? Perhaps in a followup PR?

**Comment by stanlyDoge** [comment]
Let me do it in the separate PR. I want to reduce the count of my currently opened PRs.

**Comment by scholzj** [comment]
So, I assume we need to first merge the Java 11 PR and then rebase this and remove the duplicate parts?

**Comment by stanlyDoge** [comment]
Yeah, I guess so. I am trying to improve test stability with java11. There are some race conditions in the `ClusterOperatorTest`. That makes java11 build very flaky.

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by scholzj** [comment]
Nice ... 16 tries later and it seems to work. What was the issue?

**Comment by stanlyDoge** [comment]
It seems topic operator IT suite is little bit fragile. The actual failing test was in mocked test class. And failed test does screw up the env somehow (does not stop kafka cluster). I know there will be a comment it should be fixed in this PR but I need some time to rest from this :man_shrugging:

**Comment by stanlyDoge** [comment]
@strimzi-ci run tests

**Comment by tombentley** [comment]
:clap: @stanlyDoge

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java [hunk 17]
```diff
@@ -44,6 +49,11 @@ public static void before() {
         client = new DefaultKubernetesClient();
     }
 
+    @BeforeEach
+    public void renameResource() {
+        resourceName = getResourceName(RESOURCE_NAME);
+    }
+
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java:56]: Why is this needed? Can you explain?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java [hunk 18]
```diff
@@ -40,12 +44,18 @@
...
     protected static Vertx vertx;
     protected static KubernetesClient client;
     protected static String namespace = "resource-operator-it-namespace";
 
     private static KubeClusterResource cluster;
 
+    @BeforeEach
+    public void renameResource() {
+        this.resourceName = getResourceName(RESOURCE_NAME);
+    }
```
Comment (scholzj) [inline operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractResourceOperatorIT.java:57]: Again, could you explain why this is needed?

File: operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java (hunk 17 above)
Comment (stanlyDoge) [reply operator-common/src/test/java/io/strimzi/operator/common/operator/resource/AbstractNonNamespacedResourceOperatorIT.java:56]: I was running into some flakiness of the tests because all the resources had the same name.

**Review by scholzj — APPROVED** [review]
